### ISSUE TITLE
Different imap behaviour in terminal and GUI vim

### ISSUE THREAD

**Issue by fedorenchik** [vim/vim#8799]
**Describe the bug**
When using mapping:
```
imap <C-L> ctrl-l
```
it's behaviour in terminal and GUI vim is different.

**To Reproduce**
Detailed steps to reproduce the behavior:
I) Terminal Vim:
1. Run `vim --clean`
2. Add mapping: `imap <C-L> ctrl-l`
3. In insert mode, type `<C-L>` - `ctrl-l` is inserted - as expected
4. In insert mode, type `<C-X><C-L>` - cmdline completion for line starts - as expected

II) GUI Vim:
1. Run `gvim --clean`
2. Add mapping: `imap <C-L> ctrl-l`
3. In insert mode, type `<C-L>` - `ctrl-l` is inserted - as expected
4. In insert mode, type `<C-X><C-L>` -  `ctrl-l` is inserted - **unexpected**

**Expected behavior**
Mapping in GUI vim behave the same as in terminal vim. 

**Environment (please complete the following information):**
 - Vim version: 8.2.3376
 - OS: Manjaro GNOME
 - Terminal: kitty, GTK3 GUI.

**Comment by archy3** [comment]
I noticed this since I have lots of insert mode remaps that start with `<C-f>` and all the sudden my filename completion stopped working. I found a work around was to place this within my gvimrc:
```
inoremap <C-x><C-]> <C-x><C-]>
inoremap <C-x><C-d> <C-x><C-d>
inoremap <C-x><C-e> <C-x><C-e>
inoremap <C-x><C-f> <C-x><C-f>
inoremap <C-x><C-i> <C-x><C-i>
inoremap <C-x><C-k> <C-x><C-k>
inoremap <C-x><C-l> <C-x><C-l>
inoremap <C-x><C-n> <C-x><C-n>
inoremap <C-x><C-o> <C-x><C-o>
inoremap <C-x><C-p> <C-x><C-p>
inoremap <C-x>s <C-x>s
inoremap <C-x><C-u> <C-x><C-u>
inoremap <C-x><C-v> <C-x><C-v>
inoremap <C-x><C-y> <C-x><C-y>
```

This still doesn't make the behavior completely normal. If in insert mode you press `<C-x>` and wait a bit and then press `<C-f>`, then the issue is still as the OP described, but when you press `<C-x><C-f>` quickly, it works as expected.

**Comment by brammool** [comment]
Fixed with patch 8.2.4502
